addons/account_edi/models/account_edi_document.py

```diff
@@ -248,6 +249,11 @@ def _process_documents_web_services(self, job_count=None, with_commit=True):
             if with_commit and len(jobs_to_process) > 1:
                 self.env.cr.commit()
 
+        # Re-trigger fail-safe
+        all_jobs_blocking = all(d.blocking_level for job in all_jobs for d in job[0])
+        if all_jobs_blocking:
+            # If all jobs are blocking, do not re-trigger
+            return 0
```

Comment:
```suggestion
        if documents[0].blocking_level:
            # all documents are in a pending state
            # Assume that nothing needs to be done anymore to not re-trigger to avoid infinite loop
            return 0